adoc/headers/multipointer.h

```diff
@@ -155,6 +161,7 @@ class multi_ptr {
   // Available only when: (is_decorated == false)
   operator multi_ptr<value_type, Space, access::decorated::yes>() const;
 
+  // Available only when: (Space == address_space::global_space)
   void prefetch(size_t numElements) const;
```

Comment:
I think we could support `prefetch` for both `global_space` and `generic_space` specializations, to allow for a generic pointer which is points to global memory. I was speaking with @Naghasan about this and we concluded that it could be done safely by casting to a global pointer and delegating to the global built-in if it's a valid conversion, otherwise doing nothing.

Though we don't really have a precedent in the specification for whether functions that take a `multi_ptr` to `global_space` or `local_space`, should also allow for `generic_space` or whether users should be expected to do an explicit conversion, so maybe there is a discussion to be had there.

Reply:
Implementors please review.

Reply:
I'm only trying to fix bugs in this PR.  If we want to extend this API to also support `generic_space`, should we do that as a separate PR (probably to SYCL-NEXT)?

Note that the table describing this API already lists this restriction:

> Available only when: `Space == access::address_space::global_space`.